### PR TITLE
Change reader and component names from "VII" to "METimage" and fix minor issues

### LINKED ISSUE #3202: Rename VII readers and update VII readers for new test data
It has been pointed out that the VII L1b reader no longer works for the publicly available test data, but also doesn't work for the newest EUMETSAT internal test data. The reader needs updates to work for the future versions of the data files.

Additionally, the "VII" name is discontinued and "METimage" should be used instead. This would mean the new reader names should be "metimage_l1b_nc" and "metimage_l2_nc" and the old names deprecated (see [these dictionaries](https://github.com/pytroll/satpy/blob/0beea130092f3270ae1af8a87644d47971acbbc4/satpy/readers/core/config.py#L33-L36)).

I'm assigning this to Andrea, but understand that other EUMETSAT employees may be doing the work.

Completing this work will also close my related Polar2Grid issue: https://github.com/ssec/polar2grid/issues/767

### PR DESCRIPTION
**Pull request by ameraner**
This PR 
- changes the name of the METimage instrument reader from `vii_l1b_nc` and `vii_l2_nc` to `metimage_l1b_nc` and `metimage_l2_nc`. The old reader names are put in deprecation (they still work but emit a FutureWarning). The python files and class names are adapted accordingly.
- The platform name is translated to the OSCAR names (Metop-SG-AX)
- With these fixes, the pyspectral RSR work correctly (from https://github.com/pytroll/pyspectral/pull/278 )

On top, two minor fixes are also implemented:
- the longitude arrays are wrapped to the -180, 180 range to avoid failures in resampling corner cases
- the yaml calibration configuration of one channel was faulty

... and the `rows_per_scan` attribute is added to datasets to support EWA resampling.

 - [x] Closes #3202 
 - [x] Tests added 
 - [x] Fully documented 

FYI @cesclc @TAlonglong @ludwigVonKoopa

### CODE REVIEW

**Comment by djhoese** [comment]
@ameraner @mraspaud where are we with merging this pull request? Does it need to wait for something? Are there decisions that need to be made?

**Comment by ameraner** [comment]
Hi Dave, from my side it's good to go/ready for review

File: satpy/readers/core/metimage_nc.py [hunk 1]
```diff
@@ -121,6 +125,12 @@ def get_dataset(self, dataset_id, dataset_info):
         variable = self._standardize_dims(variable)
         return variable
 
+    @staticmethod
+    def wrap_longitude(longitude_array):
+        """Wrap longitude between -180 and 180 degrees."""
+        longitude_array = ((longitude_array + 180) % 360) - 180
```
Comment (djhoese) [inline satpy/readers/core/metimage_nc.py:131]: This doesn't have to be worked on in this PR, but I think in some work I did a long time ago on another reader or maybe in pyresample I found that it was actually faster and fewer dask tasks overall to do a `where` operation on the condition of `> 180` where `lon - 360` is what is used if True. ...or whatever the correct math is.

**Review by djhoese — APPROVED** [review]
I haven't reviewed every little detail but this looks pretty good to me. It'd be great for one more review @mraspaud @pnuu @gerritholl or someone else.
